packages/frontend-shared/src/utils/weightedChoice.ts

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash'
+
+export type WeightedAlgorithm = {
+  pick: Function
+}
+
+/**
+ * Randomly choose an index from an array based on weights
+ *
+ * Based on algorithm found here: https://dev.to/trekhleb/weighted-random-algorithm-in-javascript-1pdc
+ *
+ * @param weights array of numbered weights that correspond to the indexed values
+ * @param values array of values to choose from
+ */
+const weightedChoice = (weights: number[], values: any[]) => {
```

Comment:
Do we need to validate the lengths of the arrays match?

Reply:
Yes, that would be a good idea.  I will add that check.

Reply:
Non-blocker: Validate lengths > 0